### PR TITLE
added test for failing parse of H2Kd

### LINKED ISSUE #2: Wrong normalization for "H2Kd"
``` python
In [1]: mhcnames.normalize_allele_name("H2Kd")
Out[1]: 'HLA-H*02:01'
```

### PR DESCRIPTION
**Pull request by iskandr**
Fixes https://github.com/hammerlab/mhcnames/issues/2

Problem:
```
In [1]: mhcnames.normalize_allele_name("H2Kd")
Out[1]: 'HLA-H*02:01'
```

The reason we get this crazy parse is that the helper `split_species_prefix` expects the species part of an MHC name to be separated by a dash and thus misses "H2", interpreting instead as part of the allele name for a human allele. The new implementation allows for multiple separators (not necessarily a dash) but also allows for there to be no separator if a string starts with a species prefix. 

I'm not sure how "H2Kd" then ends up being interpreted as "H0201" but I also don't want to spent too much time on this parsing code to find out. I have a replacement in mind that uses explicitly enumerated species to guide the parsing, which makes it easier to catch exceptional circumstances.

### CODE REVIEW

**Comment by tavinathanson** [comment]
@iskandr I'm finding it hard to understand how this fix fixes the issue without more comments or explanation, can you summarize? + rebase?

**Comment by iskandr** [comment]
@tavinathanson Added some more explanation in this PR and rebased off new master.

File: mhcnames/species.py [hunk 1]
```diff
@@ -81,21 +81,21 @@
         prefix_to_species_name[prefix] = species
 
 
-def split_species_prefix(name):
+def split_species_prefix(name, seps="-:_ "):
```
Comment (tavinathanson) [inline mhcnames/species.py]: Not clear why `seps="-:_ "` vs. just `seps="-"`; seems like the former could result in unexpected behavior?

**Review by tavinathanson — APPROVED** [review]
LGTM on 🍏
